src/build/__init__.py

```diff
@@ -161,6 +164,21 @@ def __init__(self, srcdir='.', config_settings=None, python_executable=sys.execu
             python_executable=python_executable,
         )
 
+    @property
+    def python_executable(self):  # type: () -> Union[bytes, Text]
+        print(self.hook.python_executable, type(self.hook.python_executable))
+        if (
+            not isinstance(self.hook.python_executable, str)
+            and not isinstance(self.hook.python_executable, bytes)
+            and not isinstance(self.hook.python_executable, unicode)
```

Comment:
I don't understand what this check accomplishes even if it'd been defined in the setter? Note that the `python_executable` can be passed as an argument to `__init__` and will bypass the `@property` entirely.

Reply:
We do not control what comes out of `pep517`, and this makes mypy happy.

Reply:
I suggest adding a comment mentioning that this is done to placate mypy because pep517 is untyped and the type of `self.hook.python_executable` is therefore unknown because it was not obvious to me or Bernát (I think).